tests/unit/test_api_urls.py

```diff
@@ -160,37 +160,89 @@ def test_api_search_wrong_parameter(self):
         self.assertEqual(rv.content_type, 'application/json')
         self.assertEqual(json_body['status'], 404)
 
-    def test_api_patch_issue_state_status(self):
+    @patch('webcompat.db.User.query')
+    def test_api_patch_issue_state_status(self, db_user_mock):
         """Patching the issue - Incompatible state and status."""
         with webcompat.app.app_context():
+
+            # mock authenticated user
+            db_user_mock.return_value.get.return_value = get_user_mock()
+            with self.app.session_transaction() as sess:
+                sess['user_id'] = 1
+
             webcompat.app.config.update(STATUSES=STATUSES)
             data = {'state': 'closed', 'milestone': 2}
             patch_data = json.dumps(data)
             rv = self.app.patch('/api/issues/1/edit', data=patch_data,
                                 environ_base=headers)
             self.assertEqual(rv.status_code, 403)
 
-    def test_api_patch_issue_too_many_json_elements(self):
+    @patch('webcompat.db.User.query')
+    def test_api_patch_issue_too_many_json_elements(self, db_user_mock):
         """Patching the issue - Too many elements in the JSON."""
         with webcompat.app.app_context():
+
+            # mock authenticated user
+            db_user_mock.return_value.get.return_value = get_user_mock()
+            with self.app.session_transaction() as sess:
+                sess['user_id'] = 1
+
             data = {'state': 'open', 'milestone': 2, 'foobar': 'z'}
             patch_data = json.dumps(data)
             rv = self.app.patch('/api/issues/1/edit', data=patch_data,
                                 environ_base=headers)
             self.assertEqual(rv.status_code, 403)
 
+    def test_api_patch_issue_without_auth(self):
+        """Patching the issue - Request without auth."""
+        with webcompat.app.app_context():
+            data = {'state': 'open', 'milestone': 2}
+            patch_data = json.dumps(data)
+            rv = self.app.patch('/api/issues/1/edit', data=patch_data,
+                                environ_base=headers)
+            self.assertEqual(rv.status_code, 403)
+
+    @patch('webcompat.db.User.query')
     @patch('webcompat.api.endpoints.api_request')
-    def test_api_patch_issue_valid_request(self, github_data):
+    def test_api_patch_issue_valid_request(self, github_data, db_user_mock):
```

Comment:
double mocks are always a bit scary. 
Because basically by faking too many things at the end we do not test anything.

Something to remember if we can improve the way we do things. so we isolate a bit more the test circumstances. To think 🤔

Reply:
Maybe we could work out something along https://github.com/webcompat/webcompat.com/blob/408d8032dcdcdd94893b2ca84e071c9b71ba7926/webcompat/helpers.py#L371